.travis.yml

```diff
@@ -42,16 +42,12 @@ git:
 
 install:
   - travis_retry ./mvnw -v
-  - |
-    if [[ -v TEST_SPECIFIC_MODULES ]]; then
-      ./mvnw install $MAVEN_FAST_INSTALL -pl "${TEST_SPECIFIC_MODULES}" -am
-    fi
   - |
     if [[ -v TEST_OTHER_MODULES ]]; then
       ./mvnw install $MAVEN_FAST_INSTALL -pl '!presto-docs,!presto-server,!presto-server-rpm'
     fi
   - |
-    if [[ -v PRODUCT_TESTS_BASIC_ENVIRONMENT || -v PRODUCT_TESTS_SPECIFIC_ENVIRONMENT || -v PRODUCT_TESTS_SPECIFIC_ENVIRONMENT_2 ]]; then
+    if [[ -v TEST_SPECIFIC_MODULES || -v PRODUCT_TESTS_BASIC_ENVIRONMENT || -v PRODUCT_TESTS_SPECIFIC_ENVIRONMENT || -v PRODUCT_TESTS_SPECIFIC_ENVIRONMENT_2 ]]; then
```

Comment:
Have you measured how much time this adds to the build? We might want to do this conditionally for the modules that need it (i.e. product tests)

Reply:
This took ` 9 hrs 28 min 22 sec`, while the current build takes `8 hrs 35 min 49 sec`

Reply:
We might want to merge builds for `kafka` and `redis` together and optimize single compilation step.

Reply:
There are two different issues:

1) Redundant work done across Travis jobs -- compiling `presto-main` and similar many times. This can be saved by merging jobs, or possibly by using stages with result sharing.
2) Unnecessary work done for specific Travis jobs -- building `presto-server` when not needed. This can be saved by not doing it.

If I'm understanding correctly, it sounds like you are saying we can make up for making `#2` worse by improving `#1`. That seems like a bad argument, because we can and should be making both better.

Instead of adding `TEST_SPECIFIC_MODULES` here, we can add a flag `INSTALL_SERVER` that is set for the modules that need it.

Reply:
> we can add a flag `INSTALL_SERVER` that is set for the modules that need it.

currently it seems we don't need an explicit flag. Building server (and cli, jdbc?) is needed whenever `MODULES_WITH_EXTRA_TESTS` is set.
if this is true, we can make it implicit for now, and make explicit only when needed